src/Phinx/Db/Adapter/SQLiteAdapter.php

```diff
@@ -155,11 +155,26 @@ public function connect()
 
             $options = $this->getOptions();
 
-            // use a memory database if the option was specified
-            if (!empty($options['memory']) || $options['name'] === static::MEMORY) {
-                $dsn = 'sqlite:' . static::MEMORY;
+            if (PHP_VERSION_ID < 80100 && (!empty($options['mode']) || !empty($options['cache']))) {
+                throw new RuntimeException('SQLite URI support requires PHP 8.1.');
+            } elseif ((!empty($options['mode']) || !empty($options['cache'])) && !empty($options['memory'])) {
+                throw new RuntimeException('Memory must not be set when cache or mode are.');
+            } elseif (PHP_VERSION_ID >= 80100) {
```

Comment:
We shouldn't be forcing everyone on PHP 8.1+ into using `mode` and `cache`. They should be able to use the existing `name` / `memory` only options if they want.

```suggestion
            } elseif (PHP_VERSION_ID >= 80100 && (!empty($options['mode']) || !empty($options['cache'])) {
```

Reply:
changed that